cicd/shared.sh

```diff
@@ -0,0 +1,306 @@
+        return
+    fi
+
+    # record an empty version so we'll get an empty table row if we don't have any versions
+    record_ext_version toolkit "$pg" ""
+
+    for ver in $TOOLKIT_VERSIONS; do
+        if [[ "$ver" = master || "$ver" = main ]]; then
+            log "skipping looking for toolkit-$ver"
+            continue
+        fi
+
+        if [ -s "$lib/timescaledb_toolkit-$ver.so" ]; then
+            #log "found toolkit-$ver for pg$pg"
+            found=true
+            record_ext_version toolkit "$pg" "$ver"
+        else
+            unsupported_reason="$(supported_toolkit "$pg" "$ver")"
+            if [ -n "$unsupported_reason" ]; then
+                log "skipped: toolkit-$ver: $unsupported_reason"
+            else
+                error "toolkit-$ver not found for pg$pg"
+            fi
+        fi
+    done
+
+    if [ "$found" = false ]; then error "no toolkit versions found for pg$pg"; fi
+}
+
+# this checks for other extensions that should always exist
+check_others() {
+    local pg="$1" lib="$2" version status
+
+    record_ext_version logerrors "$pg" ""
+    if [ -n "$PG_LOGERRORS" ]; then
+        if [ -s "$lib/logerrors.so" ]; then
+            #log "found logerrors for pg$pg"
+            record_ext_version logerrors "$pg" "$PG_LOGERRORS"
+        else
+            error "logerrors not found for pg$pg"
+        fi
+    fi
+
+    record_ext_version pg_stat_monitor "$pg" ""
+    if [ -n "$PG_STAT_MONITOR" ]; then
+        if [ -s "$lib/pg_stat_monitor.so" ]; then
+            #log "found pg_stat_monitor for pg$pg"
+            record_ext_version pg_stat_monitor "$pg" "$PG_STAT_MONITOR"
+        else
+            error "pg_stat_monitor not found for pg$pg"
+        fi
+    fi
+
+    record_ext_version pg_auth_mon "$pg" ""
+    if [ -n "$PG_AUTH_MON" ]; then
+        if [ -s "$lib/pg_auth_mon.so" ]; then
+            #log "found pg_auth_mon for pg$pg"
+            record_ext_version pg_auth_mon "$pg" "$PG_AUTH_MON"
+        else
+            error "pg_auth_mon not found for pg$pg"
+        fi
+    fi
+
+    record_ext_version postgis "$pg" ""
+    if [ -n "$POSTGIS_VERSIONS" ]; then
+        for ver in $POSTGIS_VERSIONS; do
+            IFS=\| read -rs version status <<< "$(dpkg-query -W -f '${version}|${status}' "postgresql-$pg-postgis-$ver")"
+            if [ "$status" = "install ok installed" ]; then
+                #log "found pg$pg extension postgis-$version"
+                record_ext_version postgis "$pg" "$version"
+            else
+                error "pg$pg extension postgis-$ver not found: $status"
+            fi
+        done
+    fi
+
+    for extname in $PG_WANTED_EXTENSIONS; do
+        record_ext_version "$extname" "$pg" ""
+        IFS=\| read -rs version status <<< "$(dpkg-query -W -f '${version}|${status}' "postgresql-$pg-$extname")"
+        if [ "$status" = "install ok installed" ]; then
+            #log "found pg$pg extension $extname-$version"
```

Comment:
Looks like there is a fair number of log statements which have been commented out in this script. Not sure if this is just temporary to reduce noise while debugging, or if we actually want to remove them.

Thoughts? Should we enable the commented log lines?

Reply:
`log` sends stuff to the github step summary, so we don't want to flood that. I enable for testing, but left them commented out as I've bounced between using them and not, depending on what I was troubleshooting.

I'm fine with zapping them though, as we really don't want them outputting all that extra stuff. I don't think it's terribly useful to have those statements on any output on a regular basis.

Reply:
Gone